api/src/glfs-fops.c

```diff
@@ -516,6 +516,94 @@ pub_glfs_open(struct glfs *fs, const char *path, int flags)
     return glfd;
 }
 
+GFAPI_SYMVER_PUBLIC_DEFAULT(glfs_openat, 11.0)
+struct glfs_fd *
+pub_glfs_openat(struct glfs_fd *pglfd, const char *path, int flags)
+{
+    int ret = -1;
+    struct glfs_fd *glfd = NULL;
+    xlator_t *subvol = NULL;
+    loc_t loc = {
+        0,
+    };
+    dict_t *fop_attr = NULL;
+    gf_boolean_t no_follow = 0;
+    struct iatt iatt = {0};
+
+    DECLARE_OLD_THIS;
+    __GLFS_ENTRY_VALIDATE_FD(pglfd, invalid_fs);
+
+    GF_REF_GET(pglfd);
+
+    subvol = glfs_active_subvol(pglfd->fs);
+    if (!subvol) {
+        ret = -1;
+        errno = EIO;
+        goto out;
+    }
+
+    glfd = glfs_fd_new(pglfd->fs);
+    if (!glfd)
+        goto out;
+
+    no_follow = (flags & AT_SYMLINK_NOFOLLOW) == AT_SYMLINK_NOFOLLOW;
+    glfs_lock(pglfd->fs, _gf_true);
+    {
+        ret = glfs_resolve_at(pglfd->fs, subvol, pglfd->fd->inode, path, &loc,
+                              &iatt, !no_follow, 0);
+    }
+    glfs_unlock(pglfd->fs);
+
+    if (ret < 0)
+        goto out;
+
+    if (glfd->fd) {
```

Comment:
`glfd` has been created at line 545. How can `glfd->fd` be not `NULL` ?

Reply:
Removed.